app/routes/user/endorsements.server.ts

```diff
@@ -0,0 +1,293 @@
+          '#status': 'status',
+        },
+        ExpressionAttributeValues: {
+          ':requestorSub': this.#sub,
+          ':endorserSub': escapedEndorserString,
+          ':status': 'pending',
+        },
+        FilterExpression: '#status = :status',
+      })
+    ).Items as EndorsementRequest[]
+
+    if (existingEndorsement.length > 0)
+      throw new Response(null, {
+        statusText: 'An endorsement already exists for the selected endorser',
+        status: 400,
+      })
+    const request = {
+      endorserEmail: endorserEmail,
+      endorserSub: escapedEndorserString,
+      requestorEmail: this.#currentUserEmail,
+      requestorSub: this.#sub,
+      created: Date.now(),
+      status: 'pending',
+    }
+    await db.circular_endorsements.put(request)
+  }
+
+  /**
+   * Updates the Status of an existing Endorsement Request
+   *
+   * Throws an HTTP error if:
+   *    - The current user is not in the submitters group
+   *    - There is not an existing endorsement request with a key matching the provided requestorSub, the sub of the current
+   * user as the endorserSub, and the status of that request is 'pending'.
+   * @param status - the new status to be set
+   * @param requestorSub - the sub of the user who requested the endorsement
+   */
+  async updateEndorsementRequestStatus(
+    status: EndorsementState,
+    requestorSub: string
+  ) {
+    if (
+      this.#currentUserGroups.indexOf('gcn.nasa.gov/circular-submitter') == -1
+    )
+      throw new Response(null, {
+        statusText:
+          'The user is not a verified submitter, and therefore can not approve requests',
+        status: 400,
+      })
+
+    const db = await tables()
+    await db.circular_endorsements.update({
+      Key: {
+        requestorSub: requestorSub,
+        endorserSub: this.#sub,
+      },
+      UpdateExpression: 'set #status = :status',
+      ExpressionAttributeNames: {
+        '#status': 'status',
+      },
+      ExpressionAttributeValues: {
+        ':status': status,
+      },
+      ConditionExpression: "#status = 'pending'",
+    })
+  }
+
+  /**
+   * Allows the current user to delete the existing endorsement request corresponding to
+   * the provided endorserSub.
+   * @param endorserSub - the sub of the listed endorser on the existing request
+   */
+  async deleteEndorsementRequest(endorserSub: string) {
+    const db = await tables()
+    const tableName = db.name('circular_endorsements')
+
+    const param = {
+      Key: {
+        requestorSub: this.#sub,
+        endorserSub: endorserSub,
+      },
+      TableName: tableName,
+      ConditionExpression: '#status = :pendingStatus',
+      ExpressionAttributeNames: {
+        '#status': 'status',
+      },
+      ExpressionAttributeValues: {
+        ':pendingStatus': 'pending',
+      },
+    }
+
+    db._doc.delete(param, (err, data) => {
+      if (err) throw new Response(null, { statusText: err.message })
+      return data
+    })
```

Comment:
Really? It doesn't support async/await? You have to use the old callback style?

Reply:
I guess? 
![image](https://user-images.githubusercontent.com/11023698/206539492-82a348bb-32bc-4b5d-84c9-8be2f4e25be7.png)

There appears to be a promise method on the delete function though. I'll try that out and see whats up. My implementation was just from copying the usage I came across in the docs.

Reply:
How long has it been since you rebased this branch? arc.codes recently migrated to AWS SDK v3 for the Node.js 18.x runtime. It looks like you might be using the AWS SDK v2.